sympy/matrices/matrices.py

```diff
@@ -3957,6 +3957,8 @@ def row_insert(self, pos, mti):
         row
         col_insert
         """
+        if not isinstance(pos, (int, Integer)):
```

Comment:
You can replace this with `pos = as_int(pos)` and `as_int` will raise the error if necessary.